base/doc/clsguide.tex

```diff
@@ -389,7 +390,8 @@ \subsection{Identification}
    \NeedsTeXFormat{LaTeX2e}
    \ProvidesClass{article}[2022-06-01 Standard LaTeX class]
 \end{verbatim}
-The \m{date} should be given in the form `\textsc{yyyy-mm-dd}' and must be
+The \m{date} should be given in the form `\textsc{yyyy-mm-dd}' or
+`\textsc{yyyy/nmm/dd}', and must be
 present if the optional argument is used (this is also true for the
 |\NeedsTeXFormat| command). Any derivation from this syntax will result in
```

Comment:
I think that this very detailed requirement, that there must be exactly two digits in these two locations (and four for the year) needs to explained very clearly in detail, either here or elsewhere.

Reply:
I've fixed the typo and as stated put in a note about dates

Reply:
given that some of these commands exists for 30 years (with slighlty different names) and no developer ever had a problem, my assumption is that stating it must be exactly `yyyy/mm/dd` or `yyyy-mm-dd` is precise enough.

Reply:
OK @FrankMittelbach ! But all we are doing here is making it extra clear what "exactly" means here.
This is needed because this is a "document of record" (i.e., it precisely specifies 
such rules) and not just a reminder of what has been true (and well-known) for 30 years.

Reply:
all I'm saying is that I thought stating yyyy/mm/dd is prettly clear in my opinion, but for sure with Joseph's additional footnote it is even clearer

Reply:
Yes sure: it is definitely "pretty" clear (at least to you and maybe others), but is it "sufficiently complete and accurate" for everyone?